### PR TITLE
Update dependency io.zipkin.reporter2:zipkin-reporter-bom to 3.1.1

### PR DESCRIPTION
**Pull request by codefromthecrypt**
Once merged, this will unlock spring-boot and micrometer tracing from zipkin reporter 2

### CODE REVIEW

File: dependencyManagement/build.gradle.kts [hunk 1]
```diff
@@ -16,8 +16,8 @@ val DEPENDENCY_BOMS = listOf(
   "com.squareup.okio:okio-bom:3.7.0", // applies to transitive dependencies of okhttp
   "io.grpc:grpc-bom:1.60.1",
   "io.netty:netty-bom:4.1.104.Final",
+  "io.zipkin.reporter2:zipkin-reporter-bom:3.0.2",
   "io.zipkin.brave:brave-bom:5.17.0",
-  "io.zipkin.reporter2:zipkin-reporter-bom:2.17.1",
```
Comment (codefromthecrypt) [inline dependencyManagement/build.gradle.kts:20]: note: the order matters until Brave 6 which doesn't assign a zipkin-reporter dependency version

File: exporters/zipkin/src/main/java/io/opentelemetry/exporter/zipkin/BytesEncoderAdapter.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        break;
+      case PROTO3:
+        this.encoding = Encoding.PROTO3;
+        break;
+      case THRIFT:
+        this.encoding = Encoding.THRIFT;
+        break;
+      default:
+        // Only possible if zipkin2 adds an encoding besides above, which is very unlikely.
+        throw new UnsupportedOperationException("unsupported encoding " + delegate.encoding());
```
Comment (codefromthecrypt) [inline exporters/zipkin/src/main/java/io/opentelemetry/exporter/zipkin/BytesEncoderAdapter.java:35]: this can't be unit tested as the switch is on an enum

**Comment by codefromthecrypt** [comment]
macos-latest 8 build died before starting. please kick it!

**Review by jack-berg — APPROVED** [review]
Looks good. Thanks a lot!

**Comment by codefromthecrypt** [comment]
thanks for merging!

**Pull request merged**